File: src/main/resources/output/IFC4X3_java/Entities/IfcRoot/IfcObjectDefinition/IfcTypeObject/IfcTypeProduct/IfcElementType/IfcDistributionElementType/IfcDistributionFlowElementType/IfcFlowControllerType/IfcSwitchingDeviceType/IfcSwitchingDeviceType.java
public class IfcSwitchingDeviceType extends IfcFlowControllerType {
    public IfcSwitchingDeviceTypeEnum PredefinedType;

    // === WHERE CLAUSES ===
    // CorrectPredefinedType : (PredefinedType <> IfcSwitchingDeviceTypeEnum.USERDEFINED) OR ((PredefinedType = IfcSwitchingDeviceTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcElementType.ElementType))
}

File: src/main/resources/output/IFC4X3_vbnet/Entities/IfcRoot/IfcObjectDefinition/IfcTypeObject/IfcTypeProduct/IfcElementType/IfcDistributionElementType/IfcDistributionFlowElementType/IfcFlowControllerType/IfcSwitchingDeviceType/IfcSwitchingDeviceType.vb
Public Class IfcSwitchingDeviceType Inherits IfcFlowControllerType
    Public PredefinedType As IfcSwitchingDeviceTypeEnum

    ' === WHERE CLAUSES ===
    ' CorrectPredefinedType : (PredefinedType <> IfcSwitchingDeviceTypeEnum.USERDEFINED) OR ((PredefinedType = IfcSwitchingDeviceTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcElementType.ElementType))
End Class
